rust/pact_ffi/src/models/async_message.rs

```diff
@@ -59,6 +64,41 @@ ffi_fn! {
     }
 }
 
+ffi_fn! {
+    /// Generate the message contents of an `AsynchronousMessage` as a
+    /// `MessageContents` pointer.
+    ///
+    /// This function differs from [`pactffi_async_message_get_contents`] in
+    /// that it will process the message contents for any generators or matchers
+    /// that are present in the message in order to generate the actual message
+    /// contents as would be received by the consumer.
+    ///
+    /// # Safety
+    ///
+    /// The data pointed to by the pointer must be deleted with
+    /// [`pactffi_message_contents_delete`][crate::models::contents::pactffi_message_contents_delete]
+    ///
+    /// # Error Handling
+    ///
+    /// If the message is NULL, returns NULL.
+    fn pactffi_async_message_generate_contents(message: *const AsynchronousMessage) -> *mut MessageContents {
```

Comment:
Why return a mutable pointer? What will you want to change on the contents after the call?

Reply:
The reason for the mutable pointer is that the underlying `MessageContents` is not owned by anything else, with the pointer being the only reference to it. The eventual deletion of the message contents would 'mutate' the message (in so far as deletion is a 'mutation').

I also don't see any risk in allowing people to mutate the `MessageContents`, though I also don't see a use case for that.

Reply:
Also, the `ptr::raw_to(contents)` function returns a mutable pointer, so I would either need to cast it to a non-mutable pointer; or use another function.

Reply:
In other places it casts to a const pointer, and then casts back to call the drop function. Just makes the intent clear, but is not a biggy. For instance https://github.com/pact-foundation/pact-reference/blob/1dce67a50148a2671b515dfb07aae076e08777aa/rust/pact_ffi/src/models/consumer.rs#L61